packages/rug/src/utils.js

```diff
@@ -26,10 +34,35 @@ export const calcModularScale = curry((scale, exponent) =>
   Math.round(base * Math.pow(scale, Math.round(exponent))) / base
 )
 
+/**
+ * Color Utilities
+ */
+
 export const addUnit = curry((unit, value) => `${value}${unit}`)
 
-export const opacify = (o, rgb) => rgb.replace('b(', 'ba(').replace(')', `,${o})`)
+export const opacify = (rgb, o = 1) => rgb.replace('b(', 'ba(').replace(')', `,${o})`)
+
+export const getContrastColor = ({dark, light}) => (color) => {
+  const [r, g, b] = typeof color === 'string' ? parse(color) : color
+  return (((r*299)+(g*587)+(b*114))/1000) <= 128 ? light : dark
+}
+
+const transformLab = (transform) => pipe(
+  parse,
+  rgb2lab,
+  transform,
+  lab2rgb,
+  rgb2string
+)
+
+const modL = (amount) => ([L, a, b]) => ([L += amount, a, b])
+
+export const luminosify = (color, amount = 0) => transformLab(modL(amount))(color)
+
+export const gradient = curry((type, degrees, color, lighten = false) => (
+  `${type}-gradient(${degrees}, ${lighten ? color : luminosify(color, -5)}, ${lighten ? luminosify(color, 5) : color})`
+))
 
-export const getContrastColor = ({dark, light}) => ([r, g, b]) => (
-  (((r*299)+(g*587)+(b*114))/1000) <= 128 ? light : dark
+export const shadow = (color) => (
+  `0 4px 6px ${opacify(color, 0.11)}, 0 1px 3px ${opacify(color, 0.08)}`
```

Comment:
This is not a utility. It's too specific.

I've actually got a shadow utility in my current branch I was going to push soon. We then need a shadows array for different levels.